File: 2232-adding-spaces-to-a-string/adding-spaces-to-a-string.js
/**
 * @param {string} s
 * @param {number[]} spaces
 * @return {string}
 */
var addSpaces = function(s, spaces) {
    
    let res = ""
    let space_idx = 0
    for(let i = 0; i < s.length; i++){
        if(i === spaces[space_idx]){
            const char = s[i] ? s[i] : ""
            res += " "+char
            space_idx++
        }else {
            res += s[i]
        }
    }

    return res

};

File: 2232-adding-spaces-to-a-string/adding-spaces-to-a-string.cpp
class Solution {
public:
    string addSpaces(string s, vector<int>& spaces) {
        
        string res;
        int spaces_index = 0;
        int n = spaces.size();

        for(int i = 0; i < s.length(); i++){
            if (spaces_index < n && i == spaces[spaces_index]) {
                res += " ";
                spaces_index++;
            }
            res += s[i];
        }

        return res;

    }
};